scripts/generator/graphql_generator/csharp/SDK/iOS/iOSNativeCheckout.ApplePayEventReceiver.cs.erb

```diff
@@ -41,6 +83,111 @@ namespace <%= namespace %>.SDK.iOS {
             });
         }
 
+        private static bool IsShippingAddressField(string field) {
+            return ShippingAddressFieldsSet.Contains(field);
+        }
+
+        private static bool IsContactField(string field) {
+            return ShippingContactFieldsSet.Contains(field);
+        }
+
+        private static bool IsPartialShippingAddressField(string field) {
+            return PartialShippingAddressFieldsSet.Contains(field);
+        }
+
+        private static UpdateRequestStatus GetUpdateRequestStatusFromCheckoutUserErrors(List<UserError> errors) {
+
+            var payErrors = new List<ApplePayError>();
+
+            // Check to see if any of the user errors are not billing address errors
+            // If it is not a billing address error return a failure
+            foreach (var error in errors) {
+                var fields = error.field();
+                if (fields.Contains("billingAddress")) {
+                    string description = error.message();
+                    string lastField = fields[fields.Count - 1];
```

Comment:
So here https://github.com/Shopify/unity-buy-sdk/pull/294/files#diff-dd4b7dc40c95a7f0a9eb368e629d5e53R137 we have the access in the try, so if it explodes, it reports Failure, which I think makes sense, but then in this case its not under the try, so I think it could let an out of bounds reach the user. Should this not be under the try too?